File: 1401-circle-and-rectangle-overlapping/1401-circle-and-rectangle-overlapping.py
class Solution(object):
    def checkOverlap(self, r, x, y, x1, y1, x2, y2):
        closex = max(x1 , min(x2 , x))
        closey = max(y1 , min(y2 , y))
        dist = (closex - x) * (closex - x) + (closey - y) * (closey - y)
        if dist <= r * r:
            return True
        return False

File: 1401-circle-and-rectangle-overlapping/1401-circle-and-rectangle-overlapping.cpp
class Solution {
public:
    bool checkOverlap(int r, int x, int y, int x1, int y1, int x2, int y2) {
        int closex = max(x1, min(x , x2));
        int closey = max(y1, min(y , y2));
        int dist = (closex - x) * (closex - x) + (closey - y) * (closey - y);
        if(dist <= r * r) return true;
        return false;
    }
};